pkg/model/components/nodeterminationhandler.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package components
+
+import (
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/loader"
+)
+
+// NodeTerminationHandlerOptionsBuilder adds options for cluster autoscaler to the model
+type NodeTerminationHandlerOptionsBuilder struct {
+	*OptionsContext
+}
+
+var _ loader.OptionsBuilder = &NodeTerminationHandlerOptionsBuilder{}
+
+func (b *NodeTerminationHandlerOptionsBuilder) BuildOptions(o interface{}) error {
+	clusterSpec := o.(*kops.ClusterSpec)
+	if clusterSpec.Components == nil || clusterSpec.Components.NodeTerminationHandler == nil {
```

Comment:
We should probably enable node termination handler by default for k8s 1.19+.

Reply:
I am not against that. But I suggest testing how this works in one version of kops first. So maybe enable it for 1.20+.

Reply:
Please don't enable by default.